### PR TITLE
Adding properties to TelemetrySpan

### PR DESCRIPTION
**Pull request by eddynaka**
Refs #947

## Changes

Please provide a brief description of the changes here. Update the
`CHANGELOG.md` for non-trivial changes.

For significant contributions please make sure you have completed the following items:

* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: src/OpenTelemetry.Api/Trace/Tracer.cs [hunk 1]
```diff
@@ -144,6 +144,20 @@ public TelemetrySpan StartActiveSpan(string name, SpanKind kind = SpanKind.Inter
...
+            return kind switch
+            {
+                ActivityKind.Client => SpanKind.Client,
+                ActivityKind.Consumer => SpanKind.Consumer,
+                ActivityKind.Internal => SpanKind.Internal,
+                ActivityKind.Producer => SpanKind.Producer,
+                ActivityKind.Server => SpanKind.Server,
+                _ => SpanKind.Internal,
+            };
+        }
```
Comment (eddynaka) [inline src/OpenTelemetry.Api/Trace/Tracer.cs:159]: Don't know where is the best place to add this.

**Comment by eddynaka** [comment]
@cijothomas , the only one that I didn't add was: `span.Links` vs `activity.Links`.
The issue: `activity.Links` is `IEnumerable<ActivityLink>`, and that one has many properties. Should we expose as a `IEnumerable<KeyValue...>`?

File: src/OpenTelemetry.Api/Trace/TelemetrySpan.cs [hunk 2]
```diff
@@ -60,6 +60,38 @@ public SpanContext Context
             }
         }
 
+        /// <summary>
+        /// Gets the parent span context.
+        /// </summary>
+        public SpanContext ParentSpanContext
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Api/Trace/TelemetrySpan.cs:66]: Should we let people do `ParentSpan.Context`? I think on Activity you have to do similar.

Comment (eddynaka) [reply src/OpenTelemetry.Api/Trace/TelemetrySpan.cs:66]: yeah, you are right. But, I don't know the answer.
@cijothomas , what do you think? Should we expose the property or should we let the user use `ParentSpan.Context`?

Comment (cijothomas) [reply src/OpenTelemetry.Api/Trace/TelemetrySpan.cs:66]: There are active spec discussion in this area. Lets come back to it once it settles. We can close this PR now and pick up later.

**Pull request closed without merging**